File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.java
import java.util.Stack;

class Solution {
    public String removeDuplicates(String s) {
        if (s.length() == 1) return s;

        Stack<Character> stack = new Stack<>();

        for (char curr : s.toCharArray()) {
            if (stack.isEmpty() || stack.peek() != curr) stack.push(curr);
            else stack.pop();
            
        }

    
        char[] resArray = new char[stack.size()];
        for (int i = stack.size() - 1; i >= 0; i--) {
            resArray[i] = stack.pop();
        }

        return new String(resArray);
    }
}

File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {

    if(s.length==1) return s;

    let stack = [];

    for(let curr of s){

        if(stack.length==0 || stack[stack.length-1]!==curr) stack.push(curr);

        else stack.pop();

    }

   
    return stack.join('');
    
};